### PR TITLE
[TEXT-229] Add XmlEncoderStringLookup/XmlDecoderStringLookup

### PR DESCRIPTION
**Pull request by michael-o**
This closes #449

### CODE REVIEW

**Comment by garydgregory** [comment]
Since this PR relies on `StringEscapeUtils.escapeXml10(key)` and XML 1.1 exists, shouldn't we either support both or change the names in this PR to explicitly call out XML 1.0? For example, `DefaultStringLookup.XML_DECODER` -> `DefaultStringLookup.XML_DECODER_1_0`.

**Comment by michael-o** [comment]
I'd say that 1.0 should be the default since 1.1 gained little traction. If someone wants later 1.1 support we can add, but given that no one even requested 1.0 for the past years I am reluctant to rename those coders.

File: src/test/java/org/apache/commons/text/lookup/XmlEncoderStringLookupTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link XmlEncoderStringLookup}.
+ */
+public class XmlEncoderStringLookupTest {
+
+    private static final String DATA = "&lt;element&gt;";
```
Comment (elharo) [inline src/test/java/org/apache/commons/text/lookup/XmlEncoderStringLookupTest.java:28]: inline. DATA is much less clear than the literal string. Named constants are for numbers, not strings.

File: src/test/java/org/apache/commons/text/lookup/XmlDecoderStringLookupTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link XmlDecoderStringLookup}.
+ */
+public class XmlDecoderStringLookupTest {
+
+    private static final String DATA = "<element>";
```
Comment (elharo) [inline src/test/java/org/apache/commons/text/lookup/XmlDecoderStringLookupTest.java:28]: inline. DATA is much less clear than the literal string. Named constants are for numbers, not strings.

File: src/main/java/org/apache/commons/text/lookup/XmlEncoderStringLookup.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @since 1.11.0
+ */
+final class XmlEncoderStringLookup extends AbstractStringLookup {
+
+    /**
+     * Defines the singleton for this class.
+     */
+    static final XmlEncoderStringLookup INSTANCE = new XmlEncoderStringLookup();
+
+    XmlEncoderStringLookup() {
```
Comment (elharo) [inline src/main/java/org/apache/commons/text/lookup/XmlEncoderStringLookup.java:35]: private

File: src/main/java/org/apache/commons/text/lookup/XmlDecoderStringLookup.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * @since 1.11.0
+ */
+final class XmlDecoderStringLookup extends AbstractStringLookup {
+
+    /**
+     * Defines the singleton for this class.
+     */
+    static final XmlDecoderStringLookup INSTANCE = new XmlDecoderStringLookup();
+
+    XmlDecoderStringLookup() {
```
Comment (elharo) [inline src/main/java/org/apache/commons/text/lookup/XmlDecoderStringLookup.java:35]: private

**Review by elharo — CHANGES_REQUESTED** [review]

**Comment by michael-o** [comment]
@elharo, I have copied a pre-existing class which uses the same pattern as other classes. I won't change a single class, this needs to be done for all of them in a separate PR.

**Comment by elharo** [comment]
I cry [Emerson](https://www.goodreads.com/quotes/353571-a-foolish-consistency-is-the-hobgoblin-of-little-minds-adored). Consistency with bad code is not a virtue. I see no reason to make the new API excessively complex, public, and hard-to-read.

**Comment by garydgregory** [comment]
> @elharo, I have copied a pre-existing class which uses the same pattern as other classes. I won't change a single class, this needs to be done for all of them in a separate PR.

Some constructors are private, some are not, you just happen to have picked a class for your template that had its constructor package-private instead to private.

**Comment by garydgregory** [comment]
https://issues.apache.org/jira/projects/TEXT/issues/TEXT-229

**Pull request merged**